src/Language/PureScript/Interactive.hs

```diff
@@ -291,22 +292,22 @@ handleBrowse
 handleBrowse print' moduleName = do
   st <- get
   env <- asks psciEnvironment
-  if isModInEnv moduleName st
-    then print' $ printModuleSignatures moduleName env
-    else case lookupUnQualifiedModName moduleName st of
-      Just unQualifiedName ->
-        if isModInEnv unQualifiedName st
-          then print' $ printModuleSignatures unQualifiedName env
-          else failNotInEnv moduleName
-      Nothing ->
-        failNotInEnv moduleName
+  case findMod moduleName (psciLoadedExterns st) (psciImportedModules st) of
+    Just qualName -> print' $ printModuleSignatures qualName env
+    Nothing       -> failNotInEnv moduleName
   where
-    isModInEnv modName =
-        any ((== modName) . P.getModuleName . fst) . psciLoadedExterns
-    failNotInEnv modName =
-        print' $ T.unpack $ "Module '" <> N.runModuleName modName <> "' is not valid."
-    lookupUnQualifiedModName quaModName st =
-        (\(modName,_,_) -> modName) <$> find ( \(_, _, mayQuaName) -> mayQuaName == Just quaModName) (psciImportedModules st)
+    findMod needle externs imports =
+      let
+        qualMod = fromMaybe needle (lookupUnQualifiedModName needle imports)
+        primMod = P.ModuleName [P.ProperName "Prim"]
+        modules = S.fromList (primMod : (P.getModuleName . fst <$> externs))
+       in if qualMod `S.member` modules
```

Comment:
Something seems weird with the `let ... in` indentation here. I think the `in` should be one space to the left?

Reply:
Sure! I looked for other examples of multi-line `let ... in ...`. I only found one or two, so I have emulated those. If that's not what you intended, I have no problem changing it.